src/TensorFlowNET.Core/Util/Data.cs

```diff
@@ -8,10 +9,10 @@ namespace Tensorflow.Util
     /// </summary>
     public class ValidationDataPack
     {
-        public NDArray val_x;
+        public OneOf<NDArray, NDArray[]> val_x;
```

Comment:
Is it possible to use `NDArray[]` alone since `NDArray` could also be an array of length 1? The purpose is to avoid too many APIs with `Oneof` exposed to users. If this method is not supposed to be used by our users, I think it's ok.

Reply:
The only usage of ValidationPack class is to handle the case that the validation_data parameter in model.fit could be a tuple of two NDArray or three NDArray and deconstruct to two NDArray or three NDArray. So it will not used by users.

Reply:
Done, really thank you for your advice.

Reply:
Is the usage of ValidationPack class limited to internal developers? Therefore I think we should declare the whole class as internal.

Reply:
I have tried, but in is called in model.fit, which is public, so it will cause an error : `Inconsistent accessibility: parameter type 'type' is less accessible than method 'method'` if declare it as internal.

Reply:
OK, no problem